strapi/strapi
# Allow configuration of fields during register

What it does

Modifies users-permissions to pick fields from registration body instead of omitting them and allows for plugin configuration to let users select what fields are allowed during the registration phase.

How to test

Configure the plugin like the following:

```js
// path: ./config/plugins.js

module.exports = {
  'users-permissions': {
    config: {
      register: {
        allowedFields: [ 'username', 'password', 'email', 'role' ]
      }
    }
  }
}
```

This should allow you to specify the role ID during the registration phase (not that I'd recommend that personally)

[Documentation PR](https://github.com/strapi/documentation/pull/1703)
